racket/collects/racket/private/dict.rkt

```diff
@@ -244,23 +196,39 @@
       (raise-argument-error 'dict-map "dict?" d))
     (proc (car x) (cdr x))))
```

Comment:
I could see a counter-argument, but I think `accoc-map` and `assoc-for-each` should probably also use `assoc-fold-unique`.